src/sentry/tasks/weekly_escalating_forecast.py

```diff
@@ -0,0 +1,81 @@
+from datetime import datetime
+from typing import Any, Dict, List, Tuple
+
+from django.db import transaction
+
+from sentry.issues.escalating_issues_alg import issue_spike
+from sentry.models import Group, GroupStatus
+from sentry.models.groupforecast import GroupForecast
+from sentry.models.groupsnooze import GroupSnooze
+from sentry.tasks.base import instrumented_task
+from sentry.utils.json import JSONData
+
+BATCH_SIZE = 200
+GroupForecastTuple = Tuple[Group, List[int]]
+
+
+def query_groups_past_counts(groups: List[Group]) -> JSONData:
+    return {}
+
+
+def get_forecast_per_group(
+    group_forecast_list: List[GroupForecastTuple],
+    archived_groups: List[Group],
+    response: JSONData,
+) -> None:
+    """
+    Modifies `group_forecast_list` to contain a list of forecasted values for each group.
+
+    `group_forecast_list`: List of GroupForecastTuple to be modified
+    `archived_groups`: List of archived groups to be forecasted
+    `archived_groups`: Snuba response for group event counts
+    """
+    data_index = 0
+    time = datetime.now()
+    for group in archived_groups:
```

Comment:
The way I would write this function is in two parts:

1. A function that would parse the data from Snuba
    * Iterate over the query results
    * Update a data structure that has the group IDs as the key (`group_counts[group_id] = {"intervals": [], "data": []}`)
2. A function that would generate all forecasts for every group
    * Iterate over the `group_counts` data structure and call the spike issue algorithm for each value
    * Append the group forecast
    * Return the list of group forecasts

Does this make sense to you?

It would make this function easier to read and I believe the performance would also be better since we would only iterate once over `response` once (it could be up to 360*n; n=number of groups).